boltons/cacheutils.py

```diff
@@ -450,7 +450,7 @@ def _make_cache_key(args, kwargs, typed=False, kwarg_mark=_KWARG_MARK,
 class CachedFunction(object):
     def __init__(self, func, cache, typed=False):
         self.func = func
-        self.cache = cache
+        self.cache = cache() if hasattr(cache, '__call__') else cache
```

Comment:
this should almost certainly be using [callable](https://docs.python.org/2/library/functions.html#callable)